verl/utils/model.py

```diff
@@ -688,6 +688,41 @@ def get_hf_auto_model_class(hf_config):
     return actor_module_class
 
 
+def extract_multi_modal_inputs(
+    batch_data: list[dict[str, torch.Tensor]],
+) -> dict[str, torch.Tensor | list[torch.Tensor]]:
+    """
+    Extract and process multi-modal inputs from a batch.
+
+    Args:
+        batch_data (list[dict[str, torch.Tensor]]): The batch containing potential multi-modal inputs
+
+    Returns:
+        dict[str, torch.Tensor | list[torch.Tensor]]: Processed multi-modal inputs ready for model consumption
+
+    """
+    multi_modal_inputs = {}
+    multi_modal_inputs_collected = {}
+    has_image_bound = False
+
+    for inputs in batch_data:
+        if "image_bound" in inputs:
+            has_image_bound = True
+        for key, value in inputs.items():
+            if value is not None:
+                if key not in multi_modal_inputs_collected:
+                    multi_modal_inputs_collected[key] = []
+                multi_modal_inputs_collected[key].append(value)
+
+    for key, values in multi_modal_inputs_collected.items():
```

Comment:
I think we should remove these lines. For mm models, we should not suppose these images have the same shape. After remove these lines, I can successfully train both datasets.